From 66999a149fc1181bf68e5bb4ab8288c541785cdb Mon Sep 17 00:00:00 2001
From: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Date: Wed, 18 Dec 2019 09:51:49 +0200
Subject: [PATCH 043/269] s32g: Fix compilation issue in release mode

assert() statements are not compiled unless DEBUG is passed to the build
system. This led to a warning about a variable that was used only in
debug contexts.

Upstream-Status: Pending 

Signed-off-by: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
Issue: ALB-4277
---
 plat/s32g/s32g274a_storage.c | 21 +++++++++++++++++----
 1 file changed, 17 insertions(+), 4 deletions(-)

diff --git a/plat/s32g/s32g274a_storage.c b/plat/s32g/s32g274a_storage.c
index 009486b5e..798c97704 100644
--- a/plat/s32g/s32g274a_storage.c
+++ b/plat/s32g/s32g274a_storage.c
@@ -80,23 +80,36 @@ static void plat_s32g_io_setup(enum s32g_boot_source boot_source)
 	int ret;
 
 	ret = register_io_dev_memmap(&s32g_sram_io_dev);
-	assert(ret == 0);
+	if (ret)
+		goto err_memmap;
 
 	switch (boot_source) {
 	case S32G_SRAM_BOOT:
 		ret = io_dev_open(s32g_sram_io_dev,
 				  (uintptr_t)&bl31_sram_spec,
 				  &s32g_sram_boot_dev_handle);
-		assert(ret == 0);
+		if (ret)
+			goto err_io_dev_open;
 
 		ret = io_dev_init(s32g_sram_boot_dev_handle,
 				  (uintptr_t)BL31_IMAGE_ID);
-		assert(ret == 0);
+		if (ret)
+			goto err_io_dev_init;
+
 		break;
 	default:
 		ERROR("Unknown boot source: %d", boot_source);
-		panic();
+		goto err_boot_source;
 	}
+
+	return;
+
+err_boot_source:
+err_io_dev_init:
+	io_dev_close(s32g_sram_boot_dev_handle);
+err_io_dev_open:
+err_memmap:
+	panic();
 }
 
 void s32g_io_setup(void)
-- 
2.17.1

